internal/cluster/distribution/eks/eksadapter/node_pool_manager.go

```diff
@@ -123,45 +126,91 @@ func generateNodePoolStackName(clusterName string, poolName string) string {
+		c.Status == cluster.Deleting {
+		return nil, cluster.NotReadyError{
+			OrganizationID: c.OrganizationID,
+			ID:             c.ID,
+			Name:           c.Name,
+		}
+	}
+
+	clusterClient, err := n.dynamicClientFactory.FromSecret(ctx, c.ConfigSecretID.String())
 	if err != nil {
-		return nil, errors.WrapWithDetails(err, "creating dynamic Kubernetes client factory failed", "cluster", cluster)
+		return nil, errors.WrapWithDetails(err, "creating dynamic Kubernetes client factory failed",
+			"OrganizationID", c.OrganizationID,
+			"clusterID", c.ID,
+			"clusterName", c.Name,
+			"configSecretID", c.ConfigSecretID.String(),
+		)
 	}
 
 	manager := npls.NewManager(clusterClient, n.namespace)
 	labelSets, err := manager.GetAll(ctx)
 	if err != nil {
 		return nil, errors.WrapWithDetails(err, "retrieving node pool label sets failed",
-			"cluster", cluster,
+			"OrganizationID", c.OrganizationID,
+			"clusterID", c.ID,
+			"clusterName", c.Name,
 			"namespace", n.namespace,
 		)
 	}
 
-	awsClient, err := n.awsFactory.New(cluster.OrganizationID, cluster.SecretID.ResourceID, cluster.Location)
+	awsClient, err := n.awsFactory.New(c.OrganizationID, c.SecretID.ResourceID, c.Location)
 	if err != nil {
-		return nil, errors.WrapWithDetails(err, "creating aws factory failed", "cluster", cluster)
+		return nil, errors.WrapWithDetails(err, "creating aws factory failed",
+			"OrganizationID", c.OrganizationID,
+			"clusterID", c.ID,
+			"clusterName", c.Name,
+		)
 	}
 
 	cfClient := n.cloudFormationFactory.New(awsClient)
 	describeStacksInput := cloudformation.DescribeStacksInput{}
-	nodePools := make([]eks.NodePool, 0, len(nodePoolNames))
-	for _, nodePoolName := range nodePoolNames {
-		stackName := generateNodePoolStackName(cluster.Name, nodePoolName)
-		describeStacksInput.StackName = &stackName
+	nodePools = make([]eks.NodePool, 0, len(nodePools))
+	for _, existingNodePool := range existingNodePools {
+		nodePools = append(nodePools, eks.NodePool{
+			Name:   existingNodePool.Name,
+			Labels: labelSets[existingNodePool.Name],
+		})
+		nodePool := &nodePools[len(nodePools)-1]
+
+		stackIdentifier := existingNodePool.StackID
+		if stackIdentifier == "" { // Note: CloudFormation stack creation not started yet.
+			stackIdentifier = generateNodePoolStackName(c.Name, existingNodePool.Name)
+		}
+
+		describeStacksInput.StackName = aws.String(stackIdentifier)
 		stackDescriptions, err := cfClient.DescribeStacks(&describeStacksInput)
 		if err != nil {
-			return nil, errors.WrapWithDetails(err, "retrieving node pool cloudformation stack failed",
-				"cluster", cluster,
-				"input", describeStacksInput,
-			)
+			if existingNodePool.StackID == "" {
+				// Note: node pool is in the database already, but the stack is
+				// not existing thus it is either being created or failed
+				// creation with error before CloudFormation stack creation
+				// would have been started.
```

Comment:
This branch runs if the node pool is older than this db change

Reply:
[Fixed](https://github.com/banzaicloud/pipeline/compare/f92f6f4a1158cc11e4b9a495b58a3bf3cb952702..c5954624b93bec483e9730a46e4fd9672e3a365d).